### PR TITLE
refactor gameteams first pass

### PR DESCRIPTION
**Pull request by dunlapww**
refactored rival, fave opponent, best/worst coach, and high low visitor away score.  Probably still more opportunity.

rake and rspec passing

### CODE REVIEW

File: test/game_teams_manager_test.rb [hunk 1]
```diff
@@ -26,46 +26,50 @@ def test_it_can_list_game_teams_per_coach
...
+  #   @game_teams_manager.game_teams_by_season("20142015").each do |game|
+  #     assert_instance_of GameTeam, game
+  #   end
+  #   assert_equal 8, @game_teams_manager.game_teams_by_season("20122013").count
+  #   assert_equal 24, @game_teams_manager.game_teams_by_season("20132014").count
+  #   assert_equal 32, @game_teams_manager.game_teams_by_season("20142015").count
+  #   assert_equal 10, @game_teams_manager.game_teams_by_season("20152016").count
+  #   assert_equal 14, @game_teams_manager.game_teams_by_season("20162017").count
+  #   assert_equal 18, @game_teams_manager.game_teams_by_season("20172018").count
+  # end
```
Comment (hopesgit) [inline test/game_teams_manager_test.rb:72]: Shouldn't these just be deleted?

File: lib/game_teams_manager.rb [hunk 2]
```diff
@@ -115,32 +86,16 @@ def total_score(filtered_game_teams = @game_teams)
...
-  def highest_scoring_home_team
-    highest_scoring_home_team = hoa_games_by_team_id("home").max_by do |team_id, details|
-      avg_score(details)
-    end[0]
-    @stat_tracker.fetch_team_identifier(highest_scoring_home_team)
-  end
-
-  def highest_scoring_visitor
-    highest_scoring_visitor = hoa_games_by_team_id("away").max_by do |team_id, details|
+  def scoring_team(hoa,method_arg)
```
Comment (jdcarey128) [inline lib/game_teams_manager.rb]: Can we change this name to reflect the combined methods?

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -151,27 +151,29 @@ def count_of_teams
...
 
   def lowest_scoring_home_team
-    @game_teams_manager.lowest_scoring_home_team
+    @game_teams_manager.highest_lowest_scoring_team("home",:min_by)
   end
 
   def winningest_coach(season)
-    @game_teams_manager.winningest_coach(season)
+    coach_hash = @game_teams_manager.coach_game_teams(season)
+    @game_teams_manager.highest_lowest_win_percentage(coach_hash,:max_by)
```
Comment (jdcarey128) [inline lib/stat_tracker.rb:171]: Want to refactor this with worst coach as well.

Comment (jdcarey128) [reply lib/stat_tracker.rb:171]: Nevermind

**Pull request merged**